src/main/scala/intellij/haskell/external/repl/ProjectStackReplProcess.scala

```diff
@@ -47,20 +48,24 @@ private[repl] class ProjectStackReplProcess(project: Project) extends StackReplP
   }
 
   private def load(psiFile: PsiFile, moduleName: Option[String]): Option[(StackReplOutput, Boolean)] = {
-    val filePath = getFilePath(psiFile)
-    val moduleNameAction = ApplicationManager.getApplication.runReadAction(new Computable[Option[String]] {
-      override def compute(): Option[String] = {
-        moduleName.orElse(HaskellPsiUtil.findModuleName(psiFile))
+    if (!HaskellConsoleView.isConsole(psiFile)) {
```

Comment:
Hi Rik, I just update `collectInformation` to 

```Haskell
  override def collectInformation(psiFile: PsiFile, editor: Editor, hasErrors: Boolean): PsiFile = {
    (psiFile, Option(psiFile.getOriginalFile.getVirtualFile)) match {
      case (pf, _) if HaskellConsoleView.isConsole(pf) => null
      case (_, None) => null // can be in case if file is in memory only (just created file)
      case (_, Some(f)) if f.getFileType != HaskellFileType.INSTANCE => null
      case (_, Some(_)) if HaskellProjectUtil.isLibraryFile(psiFile).getOrElse(true) => null
      case (_, Some(_)) => psiFile
    }
  }
```

But still can not prevent loading and checking console file, so my initial change seems still to be the best approach here.